crates/gitbutler-project-store/src/lib.rs

```diff
@@ -0,0 +1,92 @@
+use anyhow::{anyhow, Context, Result};
+use migrations::{gitbutler_migrations::gitbutler_migrations, migrator::Migrator};
+use rusqlite::Connection;
+use std::path::Path;
+
+mod changes;
+mod migrations;
+
+const DATABASE_NAME: &str = "project.sqlite";
+
+/// ProjectStore provides a light wrapper around a sqlite database
+struct ProjectStore {
```

Comment:
Since `Connection` is `!Sync`, i.e. `not Sync`, the whole `ProjectStore` will have to be behind a `Mutex` to compensate for that.

If `Clone` is implemented such that it connects to the database without rerunning initialization and migration then it concurrency will be possible nonetheless, and it can be as granular as `sqlite` can make it.